test/FastTests/Issues/RavenDB-12568.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Raven.Client.Documents.Commands;
+using Raven.Client.Exceptions;
+using Sparrow.Json;
+using Xunit;
+
+namespace FastTests.Issues
+{
+    public class RavenDB_12568 : RavenTestBase
+    {
+        [Fact]
+        public async Task PuttingDocumentWithMissingAttachmentsShouldThrow()
+        {
+            using (var store = GetDocumentStore())
+            {
+                var requestExecutor = store.GetRequestExecutor();
+
+                using (var context = JsonOperationContext.ShortTermSingleUse())
+                using (var stringStream = new MemoryStream(System.Text.Encoding.UTF8.GetBytes(Json)))
+                using (var blittableJson = context.Read(stringStream, "Reading of foo/bar"))
+                {
+                    try
+                    {
+                        requestExecutor.Execute(new PutDocumentCommand("foo/bar", null, blittableJson), context);
```

Comment:
Use Assert.Throws.
if this command will not throw then this test will still pass.

Reply:
It throws RavenException which is a very general exception

Reply:
Which is fine, but if it will not throw anything (e.g. someone will change sth in the future) then this test will still pass